doc/integrator/extend_application.rst

```diff
@@ -40,17 +40,243 @@ In the ``vars.yaml`` file your interface should be declared like that:
+```
+
+The panel will be included with the following HTML:
+
+```html
+<proj-feedback slot="tool-panel-feedback"></proj-feedback>
+```
+
+And panel should be an instance of:
+`gmfapi.elements.ToolPanelElement<https://camptocamp.github.io/ngeo/|main_branch|/apidoc/classes/srcapi_elements_ToolPanelElement.default.html>`_.
+
+
+In this tutorial we will create a new WebComponent based on `Lit <https://lit.dev/>`_,
+and build by `Vite <https://vitejs.dev/>`_. We will directly get the component and the build environment
+from the demo:
+
+.. prompt:: bash
+
+   cd /tmp
+   git clone git@github.com:camptocamp/demo_geomapfish.git
+   cd -
+   cp --recursive /tmp/demo_geomapfish/webcomponents \
+      /tmp/demo_geomapfish/package.json \
+      /tmp/demo_geomapfish/package-lock.json \
+      /tmp/demo_geomapfish/tsconfig.json \
+      /tmp/demo_geomapfish/vite.config.ts .
+
+Add the following lines in the ``.dockerignore``:
+
+.. code::
+
+   !webcomponents/
+   !package.json
+   !package-lock.json
+   !tsconfig.json
+   !vite.config.ts
+
+Add the following lines in the ``.gitignore``:
+
+.. code::
+
+   /node_modules
+
+Add the following lines at the end of ``Dockerfile``:
+
+.. code::
+
+   ###############################################################################
+
+   FROM node:16-slim AS custom-build
+
+   WORKDIR /app
+   COPY package.json ./
+
+   RUN npm install
+
+   COPY tsconfig.json vite.config.ts ./
+   COPY webcomponents/ ./webcomponents/
+   RUN npm run build
+
+   ###############################################################################
+
+   FROM gmf_config AS config
+   COPY --from=custom-build /app/dist/ /etc/geomapfish/static/custom/
+
+Apply the following diff in the ``geoportal/vars.yaml``:
+
+.. code:: diff
+
+   +
+   +         # For dev, the corresponding values in static should also be commented.
+   +         # gmfCustomJavascriptUrl:
+   +         #   - https://localhost:3001/@vite/client
+   +         #   - https://localhost:3001/webcomponents/index.ts
+   +
+   +         sitnFeedbackPath: custom/feedback
+   +
+   +       static:
+   +         # Those tow lines should be commented in dev mode.
+   +         gmfCustomJavascriptUrl:
+   +           name: '/etc/geomapfish/static/custom/custom.es.js'
+   +         gmfCustomStylesheetUrl:
+   +           name: /etc/geomapfish/static/css/desktop_alt.css
+   +
+   +       routes:
+   +         gmfBase:
+   +           name: base
+
+   -   content_security_policy_main_script_src_extra: "'unsafe-eval'"
+   +   content_security_policy_main_script_src_extra: "'unsafe-eval' http://localhost:3001"
+
+Working with Custom JavaScript and service
+------------------------------------------
+
+Build and run as usual:
+
+.. prompt:: bash
+
+    ./build <params>
+    docker-compose down
+    docker-compose up -d
+
+To have a development environment, with auto-reload mode you should apply the following diff in the
```

Comment:
Before this, I would like an introduction with the general principle ex:
We will run a developpement server on port ...